### PR TITLE
FSD is Based on OOPS Concepts

### PR DESCRIPTION
**Pull request by AMubinshaikh**
In this FSD-1 containing oops concepts examples
1.Abstraction
2.Polymorphism
3.Encapsulation
4.Inheritance
5.MultilevelInheritance
6.Interface
7.Method Overriding
all files have one entry point Main() method in Program.cs

### CODE REVIEW

File: Oops_Concepts/ Bank.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// is a process in which we focus display only necessary things and hide unnecessary objects
+// in other words hide implementation details from user and display only essential features
+
+using System;
+
+namespace Oops_Concepts
+{
+    public class Bank
+    {
+        private int _pin;
```
Comment (abdulmuheet) [inline Oops_Concepts/ Bank.cs]: do not use underscore( _ ) in a variable names.

File: Oops_Concepts/ Bank.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        private string _AccountHolderName;
+        public int AccountNumber = 1234567;
+        
+        public int pin { get { return _pin; } set { _pin = value; } }
+
+        public string AccountHolderName { get { return _AccountHolderName; }
```
Comment (abdulmuheet) [inline Oops_Concepts/ Bank.cs]: Correct the indentation, get{} should be in next line.

File: Oops_Concepts/ Bank.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+         // display because its necessary and public
+          public void display() 
+        {
+
+            Console.WriteLine("Customer Details name ="+ AccountHolderName+  
+                     "And account number = " + AccountNumber );
+        }
+        
+        // private doesnt show its unnecessary
+        private void Displaypin()
```
Comment (abdulmuheet) [inline Oops_Concepts/ Bank.cs]: Rename Displaypin -> DisplayPin

File: Oops_Concepts/ Bank.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        private string _AccountHolderName;
+        public int AccountNumber = 1234567;
+        
+        public int pin { get { return _pin; } set { _pin = value; } }
+
+        public string AccountHolderName { get { return _AccountHolderName; } 
+               set { _AccountHolderName = value; } }
+  
+         // display because its necessary and public
+          public void display()
```
Comment (abdulmuheet) [inline Oops_Concepts/ Bank.cs]: The method names should follow pascal casing.

File: Oops_Concepts/AddingNumbers.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+//but there is no ambiguity because we are
+// using different number of parameter or we can use different type of parameters of same function name
+
+namespace Oops_Concepts
+{
+    public class AddingNumbers
+    {   
+        // with two integer parameter
+        
+         public static int AddNumbers(int number1, int number2)
```
Comment (abdulmuheet) [inline Oops_Concepts/AddingNumbers.cs]: Why it's "static" where other one with 3 params are not?

File: Oops_Concepts/AddingNumbers.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace Oops_Concepts
+{
+    public class AddingNumbers
+    {   
+        // with two integer parameter
+        
+         public static int AddNumbers(int number1, int number2) 
+         {
+               return number1 +number2;
```
Comment (abdulmuheet) [inline Oops_Concepts/AddingNumbers.cs]: Add space after + as well.

File: Oops_Concepts/CarShowroom.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
+ //Encapsulation
+//is a process in which we can hide data from external function and 
+//external cannot access private member
+//in this example some member are public and some are private.
+using System;
+namespace Oops_Concepts
+{
+    public class CarShowroom
+    {
+    private int _CarKeyCode;
```
Comment (abdulmuheet) [inline Oops_Concepts/CarShowroom.cs]: Indentation seems incorrect.

File: Oops_Concepts/CarShowroom.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    private string _EngineCode;
```
Comment (abdulmuheet) [inline Oops_Concepts/CarShowroom.cs]: use the proper naming convention for variables.

File: Oops_Concepts/ICoachingCentre.cs [hunk 9]
```diff
@@ -0,0 +1,24 @@
+// This is an interface Containing some functions which is not implemented 
+//means function declared but withod body
```
Comment (abdulmuheet) [inline Oops_Concepts/ICoachingCentre.cs]: typo "withod" please check typing error in all the files(If any)

File: Oops_Concepts/Program.cs [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+             Console.WriteLine("The sum of two Number is = " + AddingNumbers.AddNumbers(number1,number2)); 
+
+             
+            // calling function with object because second function is not static type function
+             
+             Console.WriteLine("Enter three Numbers");
+             number1 = Convert.ToInt32(Console.ReadLine());
+             number2 = Convert.ToInt32(Console.ReadLine());
+             number3 = Convert.ToInt32(Console.ReadLine());
+             Console.WriteLine("The sum of two Number is = "
```
Comment (abdulmuheet) [inline Oops_Concepts/Program.cs:42]: It should be sum of "three" numbers...

**Review by abdulmuheet — CHANGES_REQUESTED** [review]
@AMubinshaikh Please see my comments inline.

File: Oops_Concepts/ Bank.cs (hunk 1 above)
Comment (AMubinshaikh) [reply Oops_Concepts/ Bank.cs]: ok

File: Oops_Concepts/ Bank.cs (hunk 2 above)
Comment (AMubinshaikh) [reply Oops_Concepts/ Bank.cs]: ok

File: Oops_Concepts/ Bank.cs (hunk 3 above)
Comment (AMubinshaikh) [reply Oops_Concepts/ Bank.cs]: ok

File: Oops_Concepts/ Bank.cs (hunk 4 above)
Comment (AMubinshaikh) [reply Oops_Concepts/ Bank.cs]: ok

**Review by abdulmuheet — APPROVED** [review]
Looks good

**Pull request merged**